core/src/listener/listener.ts

```diff
@@ -295,6 +314,7 @@ function historyJob({
 
     let events: ethers.Event[] = []
     try {
+      await insertUnprocessedBlocks({ blocks: [blockNumber], service })
```

Comment:
It's optional, I personally think it'd be nice if the insert functions can insert both to redis and db. and get functions try to get from redis first and fallbacks to db if not found